File: Exercise/Medium/10. Column-wise Matrix Fill/Code/Column-wise Matrix Fill.ts
function FullMatrix(n: number, arr: number[][]) {
    let num = 1;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            arr[i][j] = num++;
        }
    }
}

function PrintMatrix(n: number, arr: number[][]) {
    for (let i = 0; i < n; i++) {
        let row = "";
        for (let j = 0; j < n; j++) {
            row += arr[i][j] + " ";
        }
        console.log(row);
    }
}

function Main(n :number) {
    let arr: number[][] = new Array(n).fill([]).map(() => new Array(n).fill(0));

    FullMatrix(n, arr);
    PrintMatrix(n, arr);
    
}

Main(5);

File: Exercise/Medium/10. Column-wise Matrix Fill/Code/Column-wise Matrix Fill.cs
using System;

class Program {
    static void FullMatrix(int n, int[,] arr) {
        int num = 1;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                arr[i, j] = num++;
            }
        }
    }

    static void PrintMatrix(int n, int[,] arr) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                Console.Write(arr[i, j] + " ");
            }
            Console.WriteLine();
        }
    }

    static void Main() {
        int n = int.Parse(Console.ReadLine());

        int[,] arr = new int[n, n];

        FullMatrix(n, arr);
        PrintMatrix(n, arr);
    }
}


